src/managers/GuildChannelManager.js

```diff
@@ -50,27 +51,30 @@ class GuildChannelManager extends BaseManager {
   /**
    * Data that can be resolved to give a Guild Channel object. This can be:
    * * A GuildChannel object
+   * * A ThreadChannel object
    * * A Snowflake
-   * @typedef {GuildChannel|Snowflake} GuildChannelResolvable
+   * @typedef {GuildChannel|ThreadChannel|Snowflake} GuildChannelResolvable
    */
 
   /**
    * Resolves a GuildChannelResolvable to a Channel object.
-   * @method resolve
-   * @memberof GuildChannelManager
-   * @instance
    * @param {GuildChannelResolvable} channel The GuildChannel resolvable to resolve
-   * @returns {?GuildChannel}
+   * @returns {?GuildChannel|ThreadChannel}
    */
+  resolve(channel) {
+    if (channel instanceof ThreadChannel) return super.resolve(channel.id);
```

Comment:
Is this...needed?

Reply:
Yes, Threads don't extend GuildChannel because they are so different, but they do exist in the guild channel cache, so we resolve by id.